src/Featurizers/ShortGrainDropperFeaturizer.h

```diff
@@ -0,0 +1,255 @@
+
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Types
+    // |
+    // ----------------------------------------------------------------------
+    using GrainsSet                         = ShortGrainDropperTransformer::GrainsSet;
+    using GrainsMap                         = std::unordered_map<std::string, std::uint32_t>;
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Data
+    // |
+    // ----------------------------------------------------------------------
+    size_t const                            _colIndex;
+    std::uint32_t const                     _minPoints;
+    GrainsSet                               _grainsToDrop;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Methods
+    // |
+    // ----------------------------------------------------------------------
+    bool begin_training_impl(void) override;
+
+    // MSVC has problems when the declaration and definition are separated
+    FitResult fit_impl(typename BaseType::InputType const *pBuffer, size_t cElements) override {
+
+        InputType const * const                 pEndBuffer(pBuffer + cElements);
+
+        GrainsMap groupByGrains;
+        while(pBuffer != pEndBuffer) {
+            std::string pBufferStr = Traits<decltype(*pBuffer)>::ToString(*pBuffer);
+            GrainsMap::iterator grainsMapIter(groupByGrains.find(pBufferStr));
+            if (grainsMapIter != groupByGrains.end())
+                ++grainsMapIter->second;
+            else
+                groupByGrains.emplace(pBufferStr, 1);
+            ++pBuffer;
+        }
+
+        for (GrainsMap::value_type const & groupByGrainsElement : groupByGrains) {
+            if (groupByGrainsElement.second <= _minPoints)
+                _grainsToDrop.emplace(std::move(groupByGrainsElement.first));
+        }
+
+        return FitResult::Complete;
+    }
+
+    void complete_training_impl(void) override;
+
+    // MSVC has problems when the definition is separate from the declaration
+    typename BaseType::TransformerUniquePtr create_transformer_impl(void) override {
+        return typename BaseType::TransformerUniquePtr(new TransformerType(std::move(_grainsToDrop)));
+    }
+};
+
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// |
+// |  Implementation
+// |
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+
+// ----------------------------------------------------------------------
+// |
+// |  ShortGrainDropperTransformer
```

Comment:
>ShortGrainDropperTransformer [](start = 6, length = 28)

Move the implementation of this class into a .cpp file as it isn't a template and therefore doesn't need to be in the header file.